john-deere-api.py

```diff
@@ -0,0 +1,122 @@
+import base64
+import datetime
+import json
+import uuid
+
+from flask import Flask, render_template, request, redirect
+import requests
+import urllib.parse
+
+app = Flask(__name__)
+
+settings = {
+    'clientId': '',
+    'clientSecret': '',
+    'wellKnown': 'https://signin.johndeere.com/oauth2/aus78tnlaysMraFhC1t7/.well-known/oauth-authorization-server',
+    'callbackUrl': 'http://localhost:9090/callback',
+    'scopes': 'ag1 ag2 ag3 eq1 eq2 org1 org2 files offline_access',
+    'state': uuid.uuid1(),
+    'idToken': '',
+    'accessToken': '',
+    'refreshToken': '',
+    'apiResponse': '',
+    'accessTokenDetails': '',
+    'exp': ''
+}
+
+
+def populate(data):
+    settings['clientId'] = data['clientId']
+    settings['clientSecret'] = data['clientSecret']
+    settings['wellKnown'] = data['wellKnown']
+    settings['callbackUrl'] = data['callbackUrl']
+    settings['scopes'] = data['scopes']
+    settings['state'] = data['state']
+
+
+def update_token_info(res):
+    # settings['idToken'] = res.json()['id_token']
+    token = res.json()['access_token']
```

Comment:
In the interest of clarity, please rename `res` to something more descriptive. For simplicity, it's probably best to only call `res.json()` once and store an intermediate variable.